src-tauri/src/manager.rs

```diff
@@ -111,17 +119,36 @@ fn start_watcher(name: &'static str, tx: Sender<WatcherMessage>) {
         })
         .unwrap();
 
-        // Wait for the child to exit
-        let output = child
-            .unwrap()
-            .wait_with_output()
-            .expect("failed to wait on child");
-
-        // Send the process output to the manager
-        tx.send(WatcherMessage::Stopped {
-            name: name.to_string(),
-            output,
-        })
-        .unwrap();
+        loop {
+            match child.as_mut().unwrap().try_wait() {
+                Ok(Some(status)) => {
+                    tx.send(WatcherMessage::Stopped {
+                        name: name.to_string(),
+                        output: std::process::Output {
+                            status,
+                            stdout: Vec::new(),
+                            stderr: Vec::new(),
+                        },
+                    })
+                    .unwrap();
+                    break;
+                }
+                Ok(None) => {
+                    let mut state = state.lock().unwrap();
+                    if !state.alive {
+                        println!("Killing {name}");
+                        child.as_mut().unwrap().kill().unwrap();
```

Comment:
That's a lot of unwraps. These need to be handled.

Reply:
Will do!